common/cuda_hip/components/volatile.hpp.inc

```diff
@@ -32,13 +32,22 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 template <typename ValueType, typename IndexType>
 __device__ __forceinline__
-    std::enable_if_t<!is_complex_s<ValueType>::value, ValueType>
+    std::enable_if_t<std::is_floating_point<ValueType>::value, ValueType>
```

Comment:
does this have to do with `int` not having a definition of `is_complex`? We will probably run into issues with #1257